src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -191,197 +371,190 @@ private ExecutorInfo createExecutor(String taskIdName, String nodeName, String e
+        || taskStatus.getState().equals(TaskState.TASK_ERROR));
+  }
 
-  @Override
-  public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
-    log.info(String.format(
-        "Received status update for taskId=%s state=%s message='%s' stagingTasks.size=%d",
-        status.getTaskId().getValue(),
-        status.getState().toString(),
-        status.getMessage(),
-        liveState.getStagingTasks().size()));
+  private boolean isRunningState(TaskStatus taskStatus) {
+    return (taskStatus.getState().equals(TaskState.TASK_RUNNING));
+  }
 
-    if (status.getState().equals(TaskState.TASK_FAILED)
-        || status.getState().equals(TaskState.TASK_FINISHED)
-        || status.getState().equals(TaskState.TASK_KILLED)
-        || status.getState().equals(TaskState.TASK_LOST)) {
-      liveState.removeStagingTask(status.getTaskId());
-      liveState.removeTask(status);
-    } else if (status.getState().equals(TaskState.TASK_RUNNING)) {
-        liveState.removeStagingTask(status.getTaskId());
-        liveState.updateTask(status);
-
-        if (status.getTaskId().getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
-          if (liveState.getNameNodes().size() == HDFSConstants.TOTAL_NAME_NODES) {
-            //Finished initializing cluster after both name nodes are initialized
-            initializingCluster = false;
-          } else {
-            //Activate secondary name node after first name node is activated
-            for (TaskID taskId : liveState.getStagingTasks()) {
-              if (taskId.getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
-                sendMessageTo(driver, taskId, HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE);
-                break;
-              }
-            }
-          }
-        } else if (status.getTaskId().getValue().contains(HDFSConstants.JOURNAL_NODE_ID) &&
-              (liveState.getJournalNodes().size() ==
-              (HDFSConstants.TOTAL_NAME_NODES + conf.getJournalNodeCount()))) {
-            //Activate primary name node after all journal nodes are activated
-            for (TaskID taskId : liveState.getStagingTasks()) {
-              if (taskId.getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
-                sendMessageTo(driver, taskId, HDFSConstants.NAME_NODE_INIT_MESSAGE);
-                break;
-              }
-            }
-         }
-      }
+  private boolean isStagingState(TaskStatus taskStatus) {
+    return (taskStatus.getState().equals(TaskState.TASK_STAGING));
   }
 
-  @Override
-  public void run() {
-    FrameworkInfo.Builder frameworkInfo = FrameworkInfo.newBuilder()
-        .setName("HDFS " + conf.getClusterName())
-        .setFailoverTimeout(conf.getFailoverTimeout())
-        .setUser(conf.getHdfsUser())
-        .setRole(conf.getHdfsRole())
-        .setCheckpoint(true);
+  private void reloadConfigsOnAllRunningTasks(SchedulerDriver driver) {
+    for (Protos.TaskStatus taskStatus : liveState.getRunningTasks().values()) {
```

Comment:
Do we need to reloadConfigs on all Tasks, all Executors, or all Slaves? Just executors, I would think, since that's the scope of the sandbox, hence the config location. The way it's coded now, a JN/NN node will get two reloadConfig messages.